packages/std/src/math/uint512.rs

```diff
@@ -1174,4 +1180,25 @@ mod tests {
         assert_eq!(a.abs_diff(b), expected);
         assert_eq!(b.abs_diff(a), expected);
     }
+
+    #[test]
+    fn uint512_partial_eq() {
+        let test_cases = [(1, 1, true), (42, 42, true), (42, 24, false), (0, 0, true)]
+            .into_iter()
+            .map(|(lhs, rhs, expected)| {
+                (
+                    Uint512::from(lhs as u64),
+                    Uint512::from(rhs as u64),
```

Comment:
This creates signed ints and then unsafely casts them to u64, right? Can you use `1u64` literals instead?

Reply:
Yeah, can do, just thought this was less verbose

Reply:
I got scared of unsafe casts after seeing a some bug in a TypeScript app that popped up on the user's machine but could have been cought at compile time.

Reply:
Is there something like `Uint512::from<u64>(lhs)` for multiple versions of a trait implementation, even if `from` is not generic?